From ea27c684924b8d8a560ca6fe1f25443624d54a93 Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Wed, 18 Aug 2021 20:18:03 +0100
Subject: [PATCH] Fix some GPS receivers

Rebased version of upstream patch:
https://bazaar.launchpad.net/~foxtrotgps-team/foxtrotgps/trunk/revision/329

Conservatively added an extra API version check for gpsd.

Bug: https://bugs.gentoo.org/808883
---
 src/gps_functions.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/gps_functions.c b/src/gps_functions.c
index 602b06e..4f30cbb 100644
--- a/src/gps_functions.c
+++ b/src/gps_functions.c
@@ -762,7 +762,9 @@ cb_gpsd_data(GIOChannel *src, GIOCondition condition, gpointer data)
 		{
 			gpsdata->fix.time = (time_t) 0;
 		}
-#if GPSD_API_MAJOR_VERSION >= 9
+#if GPSD_API_MAJOR_VERSION >= 10
+		gpsdata->valid = (libgps_gpsdata.fix.status != STATUS_NO_FIX || libgps_gpsdata.fix.mode >= MODE_2D);
+#elif GPSD_API_MAJOR_VERSION >= 9
 		gpsdata->valid = (libgps_gpsdata.fix.status != STATUS_NO_FIX);
 #else
 		gpsdata->valid = (libgps_gpsdata.status != STATUS_NO_FIX);
-- 
2.33.0

